### DISCUSSION TITLE
PCI Slot changes for hot plugged devices after reboot/restart of VM

### DISCUSSION THREAD

**Discussion by saikrm04** [apache/cloudstack#9479] [Q&A]
I am using cloudstack 4.15 +KVM and I noticed that after reboot of vm the PCI slot of hot plugged nic or disk changes for windows and linux systems. Is there any change need to be done to overcome this issue?

**Comment by zap51** [comment]
> I am using cloudstack 4.15 +KVM and I noticed that after reboot of vm the PCI slot of hot plugged nic or disk changes for windows and linux systems. Is there any change need to be done to overcome this issue?
> 

Hi @saikrm04,
Would you please post your detailed observations from inside the guest? 
Let's see about the NIC first. Please also paste `virsh dumpxml <id>` on the host before vs after reboot of the guest. 

Thanks

**Reply by saikrm04** [reply]
Hi @zap51,
Here is the dumpxml before and after reboot
NIC Before Reboot
FIRST NIC
    <interface type='bridge'>
      <mac address='02:00:1e:62:00:3d'/>
      <source bridge='breth2-310'/>
      <target dev='vnet18'/>
      <model type='virtio'/>
      <driver queues='2'/>
      <link state='up'/>
      <alias name='net0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
    </interface>
HOTADD NIC 1    
    <interface type='bridge'>
      <mac address='02:00:48:42:00:04'/>
      <source bridge='breth2-313'/>
      <target dev='vnet85'/>
      <model type='virtio'/>
      <driver queues='2'/>
      <link state='up'/>
      <alias name='net1'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x07' function='0x0'/>
    </interface>


NIC After Reboot
FIRST NIC
    <interface type='bridge'>
      <mac address='02:00:1e:62:00:3d'/>
      <source bridge='breth2-310'/>
      <target dev='vnet18'/>
      <model type='virtio'/>
      <driver queues='2'/>
      <link state='up'/>
      <alias name='net0'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x03' function='0x0'/>
    </interface>
HOTADD NIC1    
    <interface type='bridge'>
      <mac address='02:00:48:42:00:04'/>
      <source bridge='breth2-313'/>
      <target dev='vnet85'/>
      <model type='virtio'/>
      <driver queues='2'/>
      <link state='up'/>
      <alias name='net1'/>
      <address type='pci' domain='0x0000' bus='0x00' slot='0x04' function='0x0'/>
    </interface>

**Reply by zap51** [reply]
The text or images you pasted is not visible.

**Reply by saikrm04** [reply]
[NIC.odt](https://github.com/user-attachments/files/16471553/NIC.odt)

**Reply by saikrm04** [reply]
Hi @zap51 Please find the text file below, somehow text is not visible. sorry for inconvenience.

**Reply by saikrm04** [reply]
Hi,
Any pointers on this.

Thanks,
sai

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
